packages/core/src/base64.ts

```diff
@@ -36,3 +36,27 @@ export function encodeBase64(data: string): string {
   }
   throw new Error('Unable to encode base64');
 }
+
+/**
+ * Encodes a string into Base64URL format.
+ * This is the encoding required for JWT parts.
+ * @param data - The unencoded input string.
+ * @returns The Base64URL encoded string.
+ */
+export function encodeBase64Url(data: string): string {
```

Comment:
This is not directly related to the external auth implementation, but it came up and it was bugging me.

There are a number of places where we generate fake JWT's in tests, and they need to be able to be decoded properly. 

In the past, we incorrectly used "base 64 encoding" rather than "base 64 URL encoding", and just got lucky that it didn't matter.

On this PR, some of the fake JWT's were hitting encoding/decoding errors, so I went ahead and added proper utilities to do this correctly.